### PR TITLE
test: Add time-timewarp-attack boundary cases

### PR DESCRIPTION
**Pull request by instagibbs**
Basic addition to test case added in https://github.com/bitcoin/bitcoin/pull/30681

### CODE REVIEW

**Comment by fjahr** [comment]
Code review ACK 31378d44f44cabc576bf92ddd61afde4b528de77

File: test/functional/mining_basic.py [hunk 1]
```diff
@@ -159,6 +159,15 @@ def test_timewarp(self):
         bad_block.solve()
         assert_raises_rpc_error(-25, 'time-timewarp-attack', lambda: node.submitheader(hexdata=CBlockHeader(bad_block).serialize().hex()))
```
Comment (tdb3) [inline test/functional/mining_basic.py:160]: I'm not immediately seeing why the previous check (for the `bad_block.nTime = t` case) is needed anymore now that the check below is being added (`bad_block.nTime = t + MAX_FUTURE_BLOCK_TIME - MAX_TIMEWARP - 1` case).  Maybe I'm overlooking something?

If not, then could add a commit that removes the check for the `bad_block.nTime = t` case.

**Review by tdb3 — APPROVED** [review]
ACK 31378d44f44cabc576bf92ddd61afde4b528de77

Great addition (testing the boundary, i.e. `MAX_FUTURE_BLOCK_TIME - MAX_TIMEWARP`).
Tested locally (and sanity checked that the assert would catch removing the `-1` from `bad_block.nTime = t + MAX_FUTURE_BLOCK_TIME - MAX_TIMEWARP - 1`).

Left one comment, but it doesn't block the ACK.

File: test/functional/mining_basic.py [hunk 2]
```diff
@@ -159,6 +159,15 @@ def test_timewarp(self):
...
 
+        self.log.info("Test timewarp protection boundary")
+        bad_block.nTime = t + MAX_FUTURE_BLOCK_TIME - MAX_TIMEWARP - 1
+        bad_block.solve()
+        assert_raises_rpc_error(-25, 'time-timewarp-attack', lambda: node.submitheader(hexdata=CBlockHeader(bad_block).serialize().hex()))
+
+        bad_block.nTime = t + MAX_FUTURE_BLOCK_TIME - MAX_TIMEWARP
```
Comment (Sjors) [inline test/functional/mining_basic.py:167]: Let's use `block` here, since it's not actually bad.

**Comment by Sjors** [comment]
Concept ACK

**Comment by achow101** [comment]
ACK 31378d44f44cabc576bf92ddd61afde4b528de77

Going to merge this now in the interest of having test coverage of the boundary before the 28.x branch off.

**Pull request merged**